specs/Picker-API_spec.md

```diff
@@ -0,0 +1,246 @@
+
+    if(0 < storageItems.Count)
+    {
+    }
+    ```
+
+## Show single folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateFolderPicker();
+    Windows.Storage.StorageFolder folder = await picker.PickSingleFolderAsync();
+    ```
+
+
+## Show multi folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateFolderPicker();
+    IReadOnlyList<Windows.Storage.StorageFolder> folders = picker.PickMultipleFoldersAsync();
+    ```
+
+# Remarks
+
+<!-- TEMPLATE
+
+    Explanation and guidance that doesn't fit into the Examples section.
+
+    APIs should only throw exceptions in exceptional conditions; basically,
+    only when there's a bug in the caller, such as argument exception.  But if for some
+    reason it's necessary for a caller to catch an exception from an API, call that
+    out with an explanation either here or in the Examples
+-->
+
+# API Notes
+
+<!-- TEMPLATE
+
+    Option 1: Give a one or two line description of each API (type and member),
+        or at least the ones that aren't obvious from their name. These
+        descriptions are what show up in IntelliSense. For properties, specify
+        the default value of the property if it isn't the type's default (for
+        example an int-typed property that doesn't default to zero.) 
+        
+    Option 2: Put these descriptions in the below API Details section,
+        with a "///" comment above the member or type. 
+-->
+
+# API Details
+
+<!-- TEMPLATE
+
+    The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/)
+    when possible, or in C# if starting with an API sketch.  GitHub's markdown
+    syntax formatter does not (yet) know about MIDL3, so use ```c# instead even
+    when writing MIDL3.
+
+    Example:
+
+    ```c# (but really MIDL3)
+    namespace Microsoft.AppModel
+    {
+        /// Represents a package on the host system. See Windows.ApplicationModel.Package for more details
+        runtimeclass Package
+        {
+            /// Returns the current package, or null if the current process is not packaged
+            static Package Current { get; };
+
+            /// Returns the package from the system store with this full name or null if not found
+            static Package GetFromFullName(String fullName);
+
+            /// Returns packages in the given family, by name
+            static Package[] FindByFamilyName(String familyName);
+        }
+    }
+    ```
+-->
+    ```c#
+    namespace Microsoft.Reunion.Storage.Pickers
+    {
+        runtimeclass StorageItemPicker
+        {
+            /// Static constructor
+            static FileOpenPicker CreateFileOpenPicker();
+            static FileSaveFileAsync CreateFileSavePicker();
+            static FolderPicker CreateFolderPicker();
+
+            String StartLocation{ set; }
+            String OkButtonText { set; }
+            String CancelButtonText { set; }
+            Windows.Foundation.Collections.IVector<String> Filters{ get; }
```

Comment:
I would appreciate it if this was changed to a type that would allow me to specify a type description, not just a file glob, as with the current Win32 file pickers. I find a raw `*.jpg` to be somewhat jarring, and much prefer `JPEG Files (*.jpg)`. This is even more important with obscure file types and/or more specific semantic requirements on what file is picked.

Reply:
Are you asking to change this type to a Map so that you are able to provide the description of the file format instead of just the extension? And the UI will populate the description + extension in the filter list instead of just the extension?

Reply:
That’s exactly correct.

Reply:
Getting the path isn't the responsibility of the picker, and not always possible for all items. Some things like the libraries (Documents, Pictures, videos, etc...) don't actually have a file system location, they are basically shell virtual folders that can contain only links to other folders. So the path for them is always empty since they don't exist as a path at the file system level. Any item that is backed by a real file system location should have a non-empty string as the Path property. This proposal should not alter that.